Cts docs clean up (#11944)
hashicorp/consul: website/content/docs/nia/compatibility.mdx

---
layout: docs
page_title: Consul-Terraform-Sync (CTS) Compatibility
description: >-
  Consul-Terraform-Sync (CTS) Compatibility
---

# Compatibility

The following tables list the Consul-Terraform-Sync (CTS) version compatibility for Consul, Terraform, and Vault.

## Consul

> Below are Consul-Terraform-Sync versions with supported Consul versions. The latest Consul-Terraform-Sync binary targets supporting the latest patch version of the three most recent Consul minor versions.

Reviewer: I'm trying to understand the second sentence here, you're saying that CTS supports the latest patch version of the three most recent Consul minor versions... I'm wondering if there might be a better way to phrase that? Maybe:

`The latest Consul-Terraform-Sync binary target supports the three most recent Consul minor versions, along with their latest patch versions.`

Reviewer: Also, and this is for the rest of the doc updates, did we want to go ahead and change the `Consul-Terraform-Sync` usage to `CTS` instead? I see you define the abbreviation at the top of the doc.

| Consul-Terraform-Sync Version | Consul OSS & Enterprise Version |
| :---------------------------- | :------------------------------ |
| CTS Enterprise 0.3+           | 1.8+                            |
| CTS OSS 0.1+                  | 1.8+                            |

## Terraform

Consul-Terraform-Sync integration with Terraform is supported for the following:

| Consul-Terraform-Sync Version | Terraform CLI Version | Terraform Cloud Version | Terraform Enterprise Version |
| :---------------------------- | :-------------------- | :---------------------- | :--------------------------- |
| CTS Enterprise 0.4+           | 0.13 - 1.1            | Latest                  | v202010-2 - Latest           |
| CTS Enterprise 0.3+           | 0.13 - 1.1            | N/A                     | v202010-2 - Latest           |
| CTS OSS 0.3+                  | 0.13 - 1.1            | N/A                     | N/A                          |
| CTS OSS 0.2                   | 0.13 - 1.0            | N/A                     | N/A                          |
| CTS OSS 0.1                   | 0.13 - 0.14           | N/A                     | N/A                          |

## Vault

> Consul-Terraform-Sync integrates with Vault to query secrets, and is supported for the following:

Reviewer: maybe might be more clear to say that `integration between CTS and Vault is support for the following:` or something that makes it clearer that it's the `integration support` that is being defined by the table?

Author: hows this `Consul-Terraform-Sync integrates with Vault to query secrets, the integration is supported for the following:` ?

# Compatibility

## Consul-Terraform-Sync

> Below are the supported Consul, Terraform, and Vault versions with compatible Consul-Terraform-Sync versions.

Reviewer 2: ```suggestion
The following table describes Consul-Terraform-Sync version compatibility for Consul, Terraform, and Vault.
```
Just trying to see if I can make this cleaner. Ignore if it doesn't seem like an improvement

> |                                                                                                    | **CTS OSS**<br/><span style={{color:'#626873'}}><sub>v0.1+</sub></span> | **CTS Enterprise**<br/><span style={{color:'#626873'}}><sub>v0.1+</sub></span> |

Reviewer 3: CTS enterprise was introduced in 0.3 + releases. Would be good to accurately represent that.

> |                                                                                                    | **CTS OSS**<br/><span style={{color:'#626873'}}><sub>v0.1+</sub></span> | **CTS Enterprise**<br/><span style={{color:'#626873'}}><sub>v0.1+</sub></span> |

Reviewer 2: Here is the only guidance I can find on use of inline styles:
https://github.com/hashicorp/engineering-docs/blob/main/writing/markdown.md#inline-styles
It doesn't address the usage in the table, but I think we'd be better off using the default table styles (left alignment, no font-color, etc.) so that we create a consistent look and feel. Also, custom styles may conflict with future design refreshes.
I would replace the checkmarks and exes with the supported version numbers from the column headers.
Also, I think we should standardize on representing versions without the 'v' in front of the number. It looks cleaner.

Author: Thanks for this context! I posted in the [#proj-consul-docs channel ](https://hashicorp.slack.com/archives/C011HP57U0Y/p1641487063001000) hoping for a response from Bryce on the styling to see if the team is looking to make an update, if not I will revert back to left alignment! I will take a look at the other feedback and make some revisions, thanks for taking the time to review!

| :------------------------------------------------------------------------------------------------: | :---------------------------------------------------------------------: | :----------------------------------------------------------------------------: |
|             **Consul OSS**<br/><span style={{color:'#626873'}}><sub>v1.8+</sub></span>             |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |
|         **Consul Enterprise**<br/><span style={{color:'#626873'}}><sub>v1.8+</sub></span>          |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

> |             **HCP Consul**<br/><span style={{color:'#626873'}}><sub>v1.8+</sub></span>             |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

Reviewer 3: Remove compatibility with HCP Consul as a row unless it has been validated by the engineering team

|                                                                                                    |
|        **Terraform OSS**<br/><span style={{color:'#626873'}}><sub>v0.13 - 1.0+</sub></span>        |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |
| **Terraform Enterprise**<br/><span style={{color:'#626873'}}><sub>v2020 10-2 - Latest</sub></span> |             <span style={{color:'#C73445'}}>&cross;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

> |    **Terraform Cloud**<br/><span style={{color:'#626873'}}><sub>Continuous Updates</sub></span>    |             <span style={{color:'#C73445'}}>&cross;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

Reviewer 3: TFC is supported for CTS Enterprise only from v0.4+

|                                                                                                    |
|            **\*Vault OSS**<br/><span style={{color:'#626873'}}><sub>v0.7+</sub></span>             |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |
|         **\*Vault Enterprise**<br/><span style={{color:'#626873'}}><sub>v0.7+</sub></span>         |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

> |            **\*HCP Vault**<br/><span style={{color:'#626873'}}><sub>v0.7+</sub></span>             |             <span style={{color:'#2EB039'}}>&check;</span>              |                 <span style={{color:'#2EB039'}}>&check;</span>                 |

Reviewer 3: Remove HCP Vault as a row unless it has been validated by the engineering team
